test/integration/utility.h

```diff
@@ -22,7 +23,10 @@ class BufferingStreamDecoder : public Http::StreamDecoder, public Http::StreamCa
   BufferingStreamDecoder(std::function<void()> on_complete_cb) : on_complete_cb_(on_complete_cb) {}
 
   bool complete() { return complete_; }
-  const Http::HeaderMap& headers() { return *headers_; }
+  const Http::HeaderMap& headers() {
+    ASSERT(headers_ != nullptr);
```

Comment:
Arguably, since we're returning a reference here, if it was nullptr the failure would be harder to understand the further away from this return we get, so best to check here.